src/mdoc/model/document.ts

```diff
@@ -88,41 +88,13 @@ export class Document {
     const namespaceRecord = this.#issuerNameSpaces.get(namespace) ?? []
 
     const addAttribute = (key: string, value: unknown) => {
-      let elementValue = value
-      if (namespace === DEFAULT_NS) {
-        // the following namespace attributes must be a full-date as specified in RFC 3339
-        if (['birth_date', 'issue_date', 'expiry_date'].includes(key) && typeof elementValue === 'string') {
-          elementValue = new DateOnly(elementValue)
-        }
-
-        if (key === 'driving_privileges' && Array.isArray(elementValue)) {
-          elementValue.forEach((v, i) => {
-            if (isObject(v) && typeof v.issue_date === 'string') {
-              // @ts-expect-error this works
-              // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-              elementValue[i].issue_date = new DateOnly(v.issue_date)
-            }
-            if (isObject(v) && typeof v.expiry_date === 'string') {
-              // @ts-expect-error this works
-              // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-              elementValue[i].expiry_date = new DateOnly(v.expiry_date)
-            }
-          })
-        }
-      }
-
       const digestID = namespaceRecord.length
       const issuerSignedItem = IssuerSignedItem.create(digestID, key, value, this.ctx)
       namespaceRecord.push(issuerSignedItem)
     }
 
     for (const [key, value] of Object.entries(values)) {
       addAttribute(key, value)
-      if (this.docType === 'org.iso.18013.5.1.mDL' && namespace === DEFAULT_NS && key === 'birth_date') {
```

Comment:
Can you add the needed checks? It's not so much work, and good to do at the same time we remove the old logic

Reply:
If we want to add validation we should add it for every field (also this validation check is incorrect, birth_date should not be a string.

I would strongly propose to add zod asap because it wouldve saved me a whole day of debugging this :/

Reply:
Okay sure let's go for zod and do it in a separate PR. Can you open an issue.

Reply:
done in #60